### PR TITLE
DT-1113: ⬆️ Upgrade kotlin to latest version and remove pinning of hi…

### PR DESCRIPTION
**Pull request by petergphillips**
…bernate and netty

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -55,18 +54,18 @@ dependencies {
   implementation(kotlin("reflect"))
 
   implementation("org.springframework.boot:spring-boot-gradle-plugin:2.3.3.RELEASE")
-  implementation("org.jetbrains.kotlin:kotlin-gradle-plugin:1.4.0")
+  implementation("org.jetbrains.kotlin:kotlin-gradle-plugin:1.4.10")
   implementation("io.spring.dependency-management:io.spring.dependency-management.gradle.plugin:1.0.10.RELEASE")
   implementation("org.owasp:dependency-check-gradle:5.3.2.1")
```
Comment (petergphillips) [inline build.gradle.kts:59]: dependency check up to 6.0 now, but not working correctly yet due to https://github.com/jeremylong/DependencyCheck/issues/2794 and https://github.com/jeremylong/DependencyCheck/issues/2804

File: src/main/kotlin/uk/gov/justice/digital/hmpps/gradle/pluginmanagers/DependencyManagementPluginManager.kt [hunk 2]
```diff
@@ -19,15 +18,4 @@ class DependencyManagementPluginManager(override val project: Project) : PluginM
       it.mavenBom(SpringBootPlugin.BOM_COORDINATES)
     }
   }
-
-  private fun forceTransitiveVersions(project: Project) {
-    val depManConfigurer = project.extensions.getByName("dependencyManagement") as DependencyManagementConfigurer
-    depManConfigurer.dependencies {
-      // Overriding 0.9.8.RELEASE introduced by Spring Boot 2.3.1 - remove when Spring Boot upgrades to > 0.9.8.RELEASE (along with tests in functional/DependencyManagementPluginManagerTest)
-      it.dependency("io.projectreactor.netty:reactor-netty:0.9.9.RELEASE")
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/gradle/pluginmanagers/DependencyManagementPluginManager.kt:27]: netty actually on 0.9.11 now

File: src/main/kotlin/uk/gov/justice/digital/hmpps/gradle/pluginmanagers/DependencyManagementPluginManager.kt [hunk 3]
```diff
@@ -19,15 +18,4 @@ class DependencyManagementPluginManager(override val project: Project) : PluginM
...
-      // Overriding 5.4.17 included by Spring Boot 2.3.1 - remove when Spring Boot upgrades to > 5.4.17.Final (along with tests in functional/DependencyManagementPluginManagerTest)
-      it.dependency("org.hibernate:hibernate-core:5.4.18.Final")
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/gradle/pluginmanagers/DependencyManagementPluginManager.kt:29]: hibernate now on 5.4.20

File: src/main/kotlin/uk/gov/justice/digital/hmpps/gradle/pluginmanagers/KotlinPluginManager.kt [hunk 4]
```diff
@@ -21,10 +21,9 @@ class KotlinPluginManager(override val project: Project) : PluginManager<KotlinP
   }
 
   private fun addDependencies() {
-    project.dependencies.add("implementation", "org.jetbrains.kotlin:kotlin-stdlib-jdk8")
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/gradle/pluginmanagers/KotlinPluginManager.kt:24]: included automatically by kotlin 1.4

**Review by matt-crowson-moj — APPROVED** [review]
LGTM

**Pull request merged**